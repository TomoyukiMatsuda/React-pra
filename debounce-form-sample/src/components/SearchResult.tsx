import React from "react";
import { IQiitaItem } from "../IQiitaItem";

interface Props {
  qiitaItems: IQiitaItem[] | null;
}

// 検索結果（フェッチしたアイテム）を表示するコンポーネント
export const SearchResult: React.VFC<Props> = (props) => {
  return (
    <div>
      {props.qiitaItems?.map((item) => {
        return (
          <div
            key={item.id}
            className="mb-3 py-2 px-8 bg-blue-100 rounded-lg shadow"
          >
            <p className="text-center font-bold  text-blue-800 mb-4 border-b-2 border-blue-800">
              {item.title}
            </p>
            <p className="text-blue-700">LGTM 👍：{item.likes_count}</p>
            <p className="text-blue-700">ユーザー：{item.user.name}</p>
          </div>
        );
      })}
    </div>
  );
};
