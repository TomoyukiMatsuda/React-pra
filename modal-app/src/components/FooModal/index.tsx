import React from "react";

interface Props {
  text: string;
}

export const FooModal: React.VFC<Props> = ({ text }) => {
  return (
    <div
      style={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        position: "fixed",
        top: "25%",
        left: "25%",
        zIndex: 20,
        height: "50%",
        width: "50%",
        boxShadow: "20px",
        background: "white",
        borderRadius: 30,
      }}
    >
      <p style={{ fontSize: 50 }}>{text}</p>
    </div>
  );
};
