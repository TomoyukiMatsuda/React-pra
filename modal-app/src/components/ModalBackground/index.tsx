import React from "react";
import Image from "next/image";

export const ModalBackground: React.VFC = () => {
  // div className="flex justify-end fixed z-10 inset-0 bg-gray-300 bg-opacity-75"
  // <button className="mt-2 mr-2 z-20 h-12 w-12 hover:bg-white hover:bg-opacity-30 rounded-full">
  return (
    <>
      <div
        style={{
          display: "flex",
          justifyContent: "end",
          position: "fixed",
          zIndex: 10,
          inset: 0,
          backgroundColor: "lightgray",
          opacity: 0.7,
        }}
      >
        <button
          style={{
            marginTop: 2,
            marginRight: 2,
            height: "3rem",
            width: "3rem",
            zIndex: 20,
            backgroundColor: "gray",
            borderRadius: "50%",
            border: "none",
            cursor: "pointer",
          }}
        >
          <Image
            src="/close_icon.svg"
            alt="close icon"
            width={30}
            height={30}
          />
        </button>
      </div>
    </>
  );
};
