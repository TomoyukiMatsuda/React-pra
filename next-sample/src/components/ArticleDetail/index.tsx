import React, { useEffect } from "react";
import { useRouter } from "next/router";
import { apiClient } from "../../lib/apiClient";

export const ArticleDetail: React.VFC = () => {
  const router = useRouter();

  useEffect(() => {
    // id が取得されないタイミングでは api を叩かない
    router.query.id &&
      apiClient
        .get(`/api/v2/items/${router.query.id}`)
        .then((response) => {
          console.log(response);
        })
        .catch((error) => {
          console.log(error);
        });
  }, [router.query.id]);

  return <div>記事ID{router.query.id}</div>;
};
