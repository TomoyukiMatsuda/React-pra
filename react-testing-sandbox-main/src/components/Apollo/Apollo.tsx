import { gql, useQuery } from "@apollo/client"

import { Pokemon, pokemonQuery } from "~/api/fetchPokemons"

export function Apollo({ size }: { size: number }) {
  const { error, loading, data } = useQuery<{ pokemons: Pokemon[] }>(gql(pokemonQuery), { variables: { size } })
  if (error) {
    throw error
  }
  if (loading || !data) {
    return <p>loading...</p>
  }
  return (
    <>
      {data.pokemons.length === 0 ? (
        <p>no pokemon</p>
      ) : (
        <ul>
          {data.pokemons.map((pokemon) => {
            return (
              <li key={pokemon.id}>
                <img src={pokemon.image} width="120" />
                <p>{pokemon.name}</p>
              </li>
            )
          })}
        </ul>
      )}
    </>
  )
}
