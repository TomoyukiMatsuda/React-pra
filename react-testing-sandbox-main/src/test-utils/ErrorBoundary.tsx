import React from "react"

export class ErrorBoundary extends React.Component<unknown, { hasError: boolean; error: Error }> {
  state = { hasError: false, error: new Error() }
  static getDerivedStateFromError(error: Error) {
    return { hasError: true, error }
  }

  render() {
    if (this.state.hasError) {
      return <div data-testid="error">{this.state.error.message}</div>
    }
    return this.props.children
  }
}
