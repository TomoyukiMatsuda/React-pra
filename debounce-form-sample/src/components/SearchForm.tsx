import React, { useState } from "react";
import Loader from "react-loader-spinner";
import { useDebounce } from "../hooks/useDebounce";
import { useFetchQiitaItems } from "../hooks/useFetchQiitaItems";
import { SearchResult } from "./SearchResult";

// 検索フォームコンポーネント（ここで debouncedValue を更新する）
export const SearchForm: React.VFC = () => {
  const [formText, setFormText] = useState<string>("");
  const debouncedValue = useDebounce(formText, 1000);
  const { qiitaItems, isLoading } = useFetchQiitaItems(debouncedValue);

  return (
    <>
      <label className="mt-12 mb-2 block text-gray-700 text-lg font-bold">
        Qiita 記事 検索キーワードを入力
      </label>
      <div className="flex w-full py-2 px-3 mb-4 shadow border rounded-full ">
        <input
          className="w-11/12 text-lg appearance-none border-none text-gray-700 focus:outline-none focus:shadow-outline"
          placeholder="例：React"
          value={formText}
          onChange={(e) => setFormText(e.target.value)}
        />
        <div className="h-6 w-6 ml-auto my-auto">
          <Loader
            type="Oval"
            color="#1E40AF"
            height={24}
            width={24}
            visible={isLoading}
          />
        </div>
      </div>
      <SearchResult qiitaItems={qiitaItems} />
    </>
  );
};
