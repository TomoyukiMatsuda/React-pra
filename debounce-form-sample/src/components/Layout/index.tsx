import React, { useState } from "react";
import { useDebounce } from "../../hooks/useDebounce";
import { useFetchQiitaItems } from "../../hooks/useFetchQiitaItems";
import { SearchResult } from "./SearchResult";
import { SearchForm } from "./SearchForm";

export const Layout: React.VFC = () => {
  const [formText, setFormText] = useState<string>("");
  const debouncedValue = useDebounce(formText, 1000);
  const { fetchQiitaItems, qiitaItems, isLoading } =
    useFetchQiitaItems(debouncedValue);

  return (
    <>
      <div className="mt-12 mb-8">
        <SearchForm
          formText={formText}
          setFormText={setFormText}
          isLoading={isLoading}
        />
      </div>
      <SearchResult fetchQiitaItems={fetchQiitaItems} qiitaItems={qiitaItems} />
    </>
  );
};
