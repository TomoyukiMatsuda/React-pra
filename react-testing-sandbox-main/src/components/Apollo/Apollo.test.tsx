import { MockedProvider, MockedResponse } from "@apollo/client/testing"
import { cleanup, render, screen, waitFor, waitForElementToBeRemoved } from "@testing-library/react"

import { Apollo } from "~/components/Apollo"
import { ErrorBoundary } from "~/test-utils/ErrorBoundary"

import { mockError, mocksSuccess, mockZero } from "./fixtures"

const WrappedApollo = ({ mocks }: { mocks: MockedResponse[] }) => {
  return (
    <ErrorBoundary>
      <MockedProvider mocks={mocks}>
        <Apollo size={2} />
      </MockedProvider>
    </ErrorBoundary>
  )
}

describe("Apollo", () => {
  afterEach(() => {
    cleanup()
  })
  test("render:loding", () => {
    const { asFragment } = render(<WrappedApollo mocks={mocksSuccess} />)
    screen.getByText("loading...")
    expect(asFragment()).toMatchSnapshot()
  })
  test("render:pokemons", async () => {
    const { asFragment } = render(<WrappedApollo mocks={mocksSuccess} />)
    await waitForElementToBeRemoved(() => screen.getByText("loading..."))
    expect(asFragment()).toMatchSnapshot()
  })
  test("render:no pokemon", async () => {
    const { asFragment } = render(<WrappedApollo mocks={mockZero} />)
    await waitFor(() => screen.getByText("no pokemon"))
    expect(asFragment()).toMatchSnapshot()
  })
  test("error", async () => {
    const spy = jest.spyOn(console, "error")
    spy.mockImplementation(() => void 0)
    render(<WrappedApollo mocks={mockError} />)
    await waitFor(() => screen.getByText("http status: 400"))
    spy.mockRestore()
  })
})
