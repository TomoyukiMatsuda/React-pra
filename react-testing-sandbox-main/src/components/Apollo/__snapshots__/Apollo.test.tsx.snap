// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apollo render:loding 1`] = `
<DocumentFragment>
  <p>
    loading...
  </p>
</DocumentFragment>
`;

exports[`Apollo render:no pokemon 1`] = `
<DocumentFragment>
  <p>
    no pokemon
  </p>
</DocumentFragment>
`;

exports[`Apollo render:pokemons 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <img
        src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
        width="120"
      />
      <p>
        Bulbasaur
      </p>
    </li>
    <li>
      <img
        src="https://img.pokemondb.net/artwork/ivysaur.jpg"
        width="120"
      />
      <p>
        Ivysaur
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
